app/agents/RecruiterAgent.py

```diff
@@ -0,0 +1,124 @@
+import spade
+from app.agents import JobOfferManagerAgent
+from app.dataaccess.model import JobOffer, Recruiter
+from app.dataaccess.model.MessageType import MessageType
+from app.modules.RecruiterModule import RecruiterModule
+from .base.BaseAgent import BaseAgent
+
+GET_STATUS_PERIOD = 20
+
+class RecruiterAgent(BaseAgent):
+    def __init__(self, recruiter_id: str, offerts_id: list[str]):
+        super().__init__(recruiter_id)  
+        self.recruiter : Recruiter = None
+        self.offerts_id = offerts_id
+        self.recruiter_id = recruiter_id
+        self.recruiterModule = RecruiterModule(self.agent_config.dbname, self.logger)
+        
+        # behaviours
+        self.get_job_offerts_stats_behav: GetStatus = None
+        self.present_analysis_behav: PresentAnalysis = None
+
+    async def setup(self):
+        await super().setup()
+        self.recruiter = self.recruiterModule.get(self.recruiter_id)
+        if self.recruiter is None:
+            self.logger.error("Recruiter information not found.")
+            await self.stop()
+
+        self.logger.info("Hello! I am representant of %s, %s.",self.recruiter.name, self.recruiter.surname)
+        self.get_job_offerts_stats_behav = GetStatus(period=GET_STATUS_PERIOD) 
+        self.add_behaviour(self.get_job_offerts_stats_behav)
+
+
+class GetStatus(spade.behaviour.PeriodicBehaviour):
+    """
+    Request to job managers to send information of the current job offerts status.
+    Activities in GAIA (role RecruiterManager): GetStatus
+    """
+
+    agent: RecruiterAgent
+
+    async def run(self):
+        if not self.agent.present_analysis_behav:
+            self.agent.logger.info("GetStatus behaviour run.")
+            for offert_id in self.agent.offerts_id:
+                prefix = self.agent.config.agents[JobOfferManagerAgent.__name__.split('.')[-1]].jid
+                self.agent.logger.info("Try to send to %s_%s", prefix, offert_id)
+                msg = await self.agent.prepare_message(
+                    f"{prefix}_{offert_id}@{self.agent.config.server.name}",
+                    "request",
+                    "status",
+                    MessageType.STATUS_REQUEST,
+                    []
+                )
+                await self.send(msg)
+                self.agent.logger.info("A message has been sent to %s_%s", prefix, offert_id)
+            self.agent.logger.info("A message has been sent to job agents requesting status.")
+        
+            self.agent.present_analysis_behav = PresentAnalysis()
+            self.agent.add_behaviour(self.agent.present_analysis_behav)
+            await self.agent.present_analysis_behav.join()
+            self.agent.logger.info("Analysis presented.")
+            self.agent.present_analysis_behav = None
+
+class PresentAnalysis(spade.behaviour.OneShotBehaviour):
```

Comment:
nie za bardzo rozumiem po nazwie, skąd to present?

Reply:
zmieniona na create